src/index.js

```diff
@@ -1,13 +1,132 @@
+// @flow
+
+import { get } from 'lodash';
+
+export type MongoOptionsType = {
+  collectionName: string,
+  timestamps: {
+    createdAt: string,
+    updatedAt: string,
+  }
+};
+
+export type MongoUserObjectType = {
+  username?: string,
+  profile?: Object,
+  services: {
+    password?: {
+      bcrypt: string,
+    },
+  },
+  emails?: [{
+    address: string,
+    verified: boolean,
+  }],
+};
+
+// TODO Will import from @accounts/account once it's published on npm
+export type UserObjectType = {
+  username: ?string,
+  email: ?string,
+  id: ?string,
+  profile: ?Object,
+  password: ?string,
+};
+
 class Mongo {
-  constructor(options) {
+  options: MongoOptionsType;
+  // TODO definition for mongodb connection object
+  db: any;
+  collection: any;
+
+  constructor(db: any, options: MongoOptionsType) {
     const defaultOptions = {
       collectionName: 'users',
+      timestamps: {
+        createdAt: 'createdAt',
+        updatedAt: 'updatedAt',
+      },
+    };
+    this.options = { ...defaultOptions, ...options };
+    if (get(db, 'constructor.define.name') !== 'Db') {
+      throw new Error('A valid database connection object is required');
+    }
+    this.db = db;
+    this.collection = this.db.collection(this.options.collectionName);
+  }
+
+  async setupIndexes(): Promise<boolean> {
+    await this.collection.createIndex('username', { unique: 1, sparse: 1 });
+    await this.collection.createIndex('emails.address', { unique: 1, sparse: 1 });
+    return true;
+  }
+
+  createUser(options: UserObjectType): Promise<UserObjectType> {
+    const user: MongoUserObjectType = {
+      services: {},
+      [this.options.timestamps.createdAt]: Date.now(),
     };
-    this.options = Object.assign({}, defaultOptions, options);
+    // TODO hash password
+    if (options.password) {
+      user.services.password = { bcrypt: options.password };
+    }
+    if (options.username) {
+      user.username = options.username;
+    }
+    if (options.email) {
+      user.emails = [{ address: options.email, verified: false }];
```

Comment:
It depends whether the data store indexes is case sensitive or not.. So the driver should handle that IMO.
Anyhow, even if we run it twice, it's a small price to pay in order to not fall on these kind of bugs in the future. I had so many projects with Meteor complaining about that..